### PR TITLE
Feature/subtask

### PR DESCRIPTION
**Pull request by MS642**
Change styling, add subtask edit and delete, use similar format for task edit and delete, change progress bar to not be dependent on reducer and remove its values from the task object, divide subtasklist to have subtask item to be able to delete and edit accordingly, and remove subtasks actions to be dependent on updatetask

TODO: change the paths to be absolute, had issues with eslinter giving me errors with absolute path

### CODE REVIEW

File: src/components/SubTask/components/SubTaskList/components/subTaskItem/index.js [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    this.state = {
+      isEditMode: false,
+      title: subTask.description,
+    };
+    this.titleInput = React.createRef();
+  }
+
+  getSubTaskIndex = (subTasks, description) => {
+    for (let i = 0; i < subTasks.length; ) {
+      if (subTasks[i].description === description) {
```
Comment (AamirL1011) [inline src/components/SubTask/components/SubTaskList/components/subTaskItem/index.js:22]: What happens if there are two subtasks with identical descriptions or is this something we don't need to worry about?

Comment (maxahn) [reply src/components/SubTask/components/SubTaskList/components/subTaskItem/index.js:22]: yeah, as Aamir mentioned we should probably add an IdObject to subtasks to uniquely identify them
also, i've discovered a findIndex function for array that will be useful for these cases

File: src/components/TaskList/index.js [hunk 2]
```diff
@@ -2,41 +2,97 @@ import React from "react";
...
 import SubTasks from "../SubTask";
+import { updateTaskItem } from "./components/TaskItem/actions";
+
+import TaskItem from "./components/TaskItem";
+
+/* To fix visual after having to use button for eslint */
+import "../SubTask/components/SubTaskList/scroll.css";
 
 class TaskList extends React.Component {
+  shouldComponentUpdate(nextProps) {
```
Comment (maxahn) [inline src/components/TaskList/index.js:23]: curious as to why this was added, isn't this default behaviour?

File: src/actions/index.js [hunk 3]
```diff
@@ -45,6 +45,8 @@ const addUserAuth = (dispatch, user) => {
       });
     })
     .catch((error) => {
-      console.error(error.message);
+      // TODO: handle error
+      return error;
+      // console.error(error.message);
```
Comment (rhiknow55) [inline src/actions/index.js:50]: uncommit bc eslint

**Pull request merged**